### PR TITLE
Bugfix in TAM container handling

### LINKED ISSUE #503: Bug in container handling for ThreadAwareModules
Currently, if a module wants to pop input containers, and store the same type of containers after processing, there is a bug that means no containers would be stored.

The relevant functionality is implemented as follows at the moment:

```
        with ThreadPoolExecutor(max_workers=module.GetThreadPoolSize())  executor:
          for container in GetContainers(module.GetThreadOnContainerType()):
            futures.append(submit(module.Process, container))
<snip>
        if not module.KeepThreadedContainersInState():
          self.GetContainers(module.GetThreadOnContainerType(), True)
```

If the module wants to pop it's input containers, they are removed after `Process`. However, if a module (for example, `SCPExporter`) uses the same container type as input and output, then they are added in Process, then removed afterwards.

### PR DESCRIPTION
**Pull request by ramo-j**
Correction & tests for an edge case in container handling for ThreadAwareModules. 

Closes #503

### CODE REVIEW

File: tests/test_modules/thread_aware_modules.py [hunk 1]
```diff
@@ -102,3 +102,37 @@ def PreProcess(self) -> None:
...
+
+  As described by https://github.com/log2timeline/dftimewolf/issues/503 this
+  module pops containers for input, and uses the same container type as output.
+  """
+  def __init__(self, state, name=None):
+    super(Issue503Module, self).__init__(state, name)
+
+  def SetUp(self): # pylint: disable=arguments-differ
+    """SetUp"""
+    self.logger.info('{0:s} SetUp!'.format(self.name))
```
Comment (itsmvd) [inline tests/test_modules/thread_aware_modules.py:117]: Prefer using f-strings over format strings, they're faster and their syntax is easier to read. Same for other format strings in this file.

**Review by itsmvd — CHANGES_REQUESTED** [review]

File: tests/test_modules/thread_aware_modules.py (hunk 1 above)
Comment (itsmvd) [reply tests/test_modules/thread_aware_modules.py:117]: As discussed in PM since this is all over this file, lets change this in another PR.

Comment (ramo-j) [reply tests/test_modules/thread_aware_modules.py:117]: Raised issue #505 for wider discussion on this.

**Review by itsmvd — APPROVED** [review]
